src/components/fast_clock.c

```diff
@@ -0,0 +1,253 @@
+  fast_clock_status.id2 = id2;
+  fast_clock_status.intermessage_delay = intermessage_delay;
+}
+
+// ----------------------------------------------------------------------------
+// sets the thing as a slave.
+void fast_clock_set_as_slave(void)
+{
+  fast_clock_status.master = false;
+}
+
+//----------------------------------------------------------------------------
+void fast_clock_set_time(FAST_CLOCK_TIME_Type time)
+{
+  // Set the time
+  current_time = time;
+  // Reset the millisecond counter
+  fast_clock_status.millisecond = 0;
+  // Notify the update!
+  fast_clock_handle_update(current_time);
+}
+
+//----------------------------------------------------------------------------
+// this function sets the clock rate.
+void fast_clock_set_rate(uint8_t rate)
+{
+  fast_clock_status.rate = rate;
+}
+
+//----------------------------------------------------------------------------
+// this function processes the clock information sent on the loconet
+void loconet_rx_fast_clock(uint8_t *data, uint8_t length)
+{
+  if (length < 8)
+  {
+    return;
+  }
+
+  if (!(data[7] == 1)) {
+    // the message is not a correct clock tick!
+    return;
+  }
+
+  //1st byte of data is the clock rate
+  fast_clock_set_rate(data[0]);
+
+  current_time.minute = data[3] - (128-60);
+  current_time.hour = data[5] >= (128-24) ? data[5] - (128-24) : data[5] % 24;
+  current_time.day = data[6] % 7;
+  current_time.second = 0;
+
+  fast_clock_status.millisecond = 0;
+
+  fast_clock_handle_update(current_time);
+}
+
+//-----------------------------------------------------------------------------
+// sends the message in the appropriate format
+static void fast_clock_send_message(void)
+{
+  loconet_tx_fast_clock(
+    fast_clock_status.rate,
+    0x0,
+    0x0,
+    current_time.minute,
+    current_time.hour,
+    current_time.day,
+    fast_clock_status.id1,
+    fast_clock_status.id2
+  );
+}
+
+
+// ----------------------------------------------------------------------------
+// this function should be called every 50ms. It updates the clock with
+// the set rate.
+void fast_clock_tick(void)
+{
+  // keeps track whether we need an update. We only send an update
+  // if the minute counter has been increased.
+  bool notify = false;
+
+  // update the milliseconds passed with the clock rate
+  fast_clock_status.millisecond += fast_clock_status.rate;
+
+  // as the millisecond counter counts every 50ms, we have 200 cycles for a
+  // second. Thus, if millisecond > 200, we can update the current_time
+  // as the rate cannot be larger than 127, we do not need fancy checks :)
+
+  if (fast_clock_status.millisecond > 200) {
+    fast_clock_status.millisecond -= 200;
+    current_time.second++;
+  }
+
+  if(current_time.second > 59)
+  {
+    current_time.minute++;
+    current_time.second = 0;
+    notify = true;
+  }
+
+  if (current_time.minute > 59)
+  {
+    current_time.minute = 0;
+    current_time.hour++;
+  }
+  if (current_time.hour > 23)
```

Comment:
Between second and minute you have a newline, between minute and hour not. Please make it consistent, either way 👍